### ISSUE TITLE
hang in ud_mlx5/uct_p2p_am_test.am_short/1 with valgrind

### ISSUE THREAD

**Issue by yosefe** [openucx/ucx#1196]
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2479/label=hpc-test-node/console
```
12:15:41 [       OK ] ud_mlx5/uct_p2p_am_test.am_sync/1 (1766 ms)
12:15:41 [ RUN      ] ud_mlx5/uct_p2p_am_test.am_short/1
12:15:41 [     INFO ] 8 make: *** [test_valgrind] Killed
13:31:48 
Build step 'Execute shell' marked build as failure
```

**Comment by yosefe** [comment]
also in http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2480/label=hpc-test-node2/console

**Comment by brminich** [comment]
Another hang in UD with valgrind 
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2482/label=hpc-test-node/console

14:25:28 [----------] 28 tests from ud/uct_p2p_am_test
14:25:28 [ RUN      ] ud/uct_p2p_am_test.am_bcopy_keep_data/1
14:25:28 [     INFO ] 8 ==27138== Gdb request to kill this process
15:53:41 make: Leaving directory `/scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node/build-test/test/gtest'

**Comment by yosefe** [comment]
yep
this should solve it: https://github.com/openucx/ucx/pull/1190/commits/753d5e4f2a4b47a69d4341299808e790e6a624bb

**Comment by brminich** [comment]
OK, will wait for that PR to be merged before retesting the failed one.

**Comment by yosefe** [comment]
fixed by #1190
